soccer/gameplay/plays/offense/distraction1.py

```diff
@@ -0,0 +1,188 @@
+
+        self.add_transition(distraction1.State.cross,
+                        distraction1.State.shoot, lambda: self.subbehavior_with_name('pass to striker').is_done_running()
+                        , 'crossing-shooting')
+
+        self.add_transition(distraction1.State.shoot,
+                        distraction1.State.setup, lambda: self.has_subbehavior_with_name('shooting') and self.subbehavior_with_name('shooting').is_done_running(), 'repeat')
+
+        
+        self.d1 = robocup.Point(0.40*constants.Field.Width, 0.95*constants.Field.Length) #the first distraction point
+        self.d2 = robocup.Point(0.40*constants.Field.Width, 0.8*constants.Field.Length) #the second distraction point
+        self.s1 = robocup.Point(-0.40*constants.Field.Width, 0.9*constants.Field.Length) #striker's position
+        self.center = robocup.Point(0,4.5) #center of field position, used if ball is far
+        
+
+    def on_enter_setup(self):
+        #capture ball and get striker and distractor in position
+        print("entered setup")
+        self.add_subbehavior(skills.capture.Capture(), 'capture1')
+        self.add_subbehavior(skills.move.Move(self.s1), 'striker moves')
+        self.add_subbehavior(skills.move.Move(self.d2), 'distract moves')
+        #if the ball is too far away to pass across the field
+        self.ball_is_far = main.ball().pos.y < 4
+        
+
+
+    def on_exit_setup(self):
+        print("exits setup")
+        self.remove_all_subbehaviors()
+        
+
+    def on_enter_optionaladjustment(self):
+        #if the ball is too far then the distractor moves to the center
+        print("entered setup2")
+        self.add_subbehavior(skills.capture.Capture(), 'capture 2')
+        self.add_subbehavior(skills.move.Move(self.center), 'move half')
+        self.add_subbehavior(skills.move.Move(self.s1), 'stay')
+        
+
+    def on_exit_optionaladjustment(self):
+        print('exits optional adjustment')
+        self.remove_all_subbehaviors()
+
+    def on_enter_centerpass(self):
+        #pass the ball to the robot in the center and move a robot to the distract position 
+        self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.center), 'center pass')
+        self.add_subbehavior(skills.move.Move(self.d2), "move back to distract")
+
+
+    def on_exit_centerpass(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_passing(self):
+        #either pass to striker or distracter depending on shot chance
+        print('entered passing')
+        
+        
+        passchance1 = evaluation.passing.eval_pass( main.ball().pos, self.d2, main.our_robots() )
+        passchance2 = evaluation.passing.eval_pass( main.ball().pos, self.s1, main.our_robots() )
+        passchance3 = evaluation.passing.eval_pass( self.d2, self.s1, main.our_robots() )
+        shotchance1 = evaluation.shooting.eval_shot( self.s1, main.our_robots() )
```

Comment:
Only `passchance1`, `passchance2` and `shotchance1` are being used. This algorithm is pretty heavy so any extra calls should be avoided if possible